libfreerdp/core/nla.c

```diff
@@ -1620,24 +1735,27 @@ static size_t nla_write_ts_password_creds(rdpNla* nla, wStream* s)
 	return size;
 }
 
-static BOOL nla_read_ts_credentials(rdpNla* nla, PSecBuffer ts_credentials)
+static BOOL nla_read_ts_credentials(rdpNla* nla, SecBuffer* data, size_t offset)
 {
 	wStream* s;
 	size_t length;
 	size_t ts_password_creds_length = 0;
-	BOOL ret;
+	BOOL ret = FALSE;
 
-	if (!ts_credentials || !ts_credentials->pvBuffer)
+	if (!data)
 		return FALSE;
 
-	s = Stream_New(ts_credentials->pvBuffer, ts_credentials->cbBuffer);
+	s = Stream_New(data->pvBuffer, data->cbBuffer);
 
```

Comment:
In this function, we could probably use a static `wStream` for `s` and so avoid the burden of freeing